### ISSUE TITLE
API Design Bug: API inconsistency between RestClient::Request.execute() and RestClient.post() 

### ISSUE THREAD

**Issue by iMerica** [rest-client/rest-client#392]
Hey guys!

I have an app with hundreds of different calls to `RestClient.post()`. All has been fine because it's been talking to a host with a proper SSL, but now I need it to talk to a local hostname with it's own SSL created using `openssl`. 

When I point it to my localhost I get ` #<RestClient::SSLCertificateNotVerified:....>`. So it would be great if I could just pass in `:verify_ssl => false` to `RestClient.post()`. 

Of course, I can do:
```ruby
client = RestClient::Resource.new('https://example.com/',
                                  :ssl_client_cert => p12.certificate,
                                  :ssl_client_key => p12.key,
                                  :verify_ssl => OpenSSL::SSL::VERIFY_NONE)
```
but `client` has a totally different API. I'd have to re-write lots of code to make it work.

So in summary, as user of your client I would prefer one really awesome and flexible API that does everything vs. multiple API styles (ActiveResource-Style vs. other styles) that can do different things.

Best,

Michael

**Comment by ab** [comment]
Hey there,

This is indeed a long standing known design flaw. I'm planning to address it in a 3.0 release at the earliest.

In general, I'd recommend always using `RestClient::Request.execute(:url => foo, :method => :POST, ...)`, which gives you access to the full API.

Cheers,
Andy

**Comment by iMerica** [comment]
For anyone coming back to this issue from a Google search: You can get around this by monkey patching the `RestClient` module. 

For example:
```ruby
module RestClient

  def self.get(url, headers={}, &block)
    Request.execute(
        :method => :get,
        :url => url,
        :headers => headers,
        :verify_ssl => false,
        &block
    )
  end

 # ... other HTTP verbs you want to patch ( e.g post, patch, delete)

end
```

Then in your client code you can do
```ruby
RestClient.get("https://myapp.dev", param1, param2) 
```
like normal without rest-client complaining about the SSL.

:information_desk_person: 
Be advised that this is not a good idea if you're sending/receiving sensitive data because any spoofing attempts will go through unchecked. This is just a workaround for people that need to connect to an otherwise safe host that doesn't have it's SSL verified.

**Comment by ab** [comment]
That works, but there's no need to monkey patch. I'd recommend using the full API here like so:

```
RestClient::Request.execute(method: :GET, url: "https://myapp.dev", verify_ssl: false, headers: ...)
```

**Comment by iMerica** [comment]
Good to know. Thanks @ab !
